yaksh/models.py

```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
         default='inprogress'
     )
 
+    # set question order
+    questions_order = models.CharField(max_length=255, blank=True, default='')
+
     objects = AnswerPaperManager()
 
     def current_question(self):
         """Returns the current active question to display."""
         if self.questions_unanswered.all():
-            return self.questions_unanswered.all()[0]
-        return self.questions.all()[0]
+            cur_question = self.get_current_question(
+                self.questions_unanswered.all())
+        else:
+            cur_question = self.get_current_question(self.questions.all())
+        return cur_question
+
+    def get_current_question(self, questions):
+        if self.questions_order:
+            question_id = int(self.questions_order.split(',')[0])
```

Comment:
This logic is not clear.  Why is the first question in the questions_order the current one?

Reply:
Isn't there some other database entry that tells you which is the active question?

Reply:
No we don't store the current question. we just take the first question from the list of questions.
Then for next question we just iterate over that list based on current question id.